service/src/test/java/org/sunbird/learner/actors/skill/UserSkillManagementActorTest.java

```diff
@@ -315,20 +322,28 @@ private Response createGetUserFailureResponse() {
     return response;
   }
 
-  private void mockGetSkillResponse(String userId) {
+  private void mockGetSkillResponse(String userId, boolean isFailure) {
     HashMap<String, Object> esDtoMap = new HashMap<>();
+
     Map<String, Object> filters = new HashMap<>();
+    Map<String, Object> response = null;
+
     filters.put(JsonKey.USER_ID, userId);
     esDtoMap.put(JsonKey.FILTERS, filters);
     List<String> fields = new ArrayList<>();
     fields.add(JsonKey.SKILLS);
     esDtoMap.put(JsonKey.FIELDS, fields);
     PowerMockito.mockStatic(ElasticSearchUtil.class);
+
+    if (!isFailure) {
+      response = createGetSkillResponse();
+    } else response = new HashMap<>();
+
     when(ElasticSearchUtil.complexSearch(
             ElasticSearchUtil.createSearchDTO(esDtoMap),
             ProjectUtil.EsIndex.sunbird.getIndexName(),
             ProjectUtil.EsType.user.getTypeName()))
-        .thenReturn(createGetSkillResponse());
+        .thenReturn(response);
```

Comment:
Within when (mocking), use matchers - not exact values. 
May not work sometimes - because you are expecting exact objects, rather than the matchers.